internal/service/cognitoidp/user_pool_domain.go

```diff
@@ -60,6 +62,10 @@ func ResourceUserPoolDomain() *schema.Resource {
 				Computed: true,
 			},
 		},
+		CustomizeDiff: customdiff.ForceNewIfChange("certificate_arn", func(_ context.Context, old, new, meta interface{}) bool {
```

Comment:
not needed if you removed `ForceNew `

Reply:
So I added this `ForceNewIfChange` to prevent the update function being ran in these two specific edge cases, which result in an error from the AWS API:
1. A custom domain was initially set with a certificate arn, then was removed (this triggers an update to set the certificate ARN to blank, which will cause an error)
2. A custom domain was not initially set with a certificate arn, then one is added (this indicates that there wasn't a custom domain, which means it must be fully destroyed and re-created to add the cert arn)

If there is an easier way to do this outside of the `ForceNewIfChange`, or if we don't think it's necessary, I can remove it.

Reply:
@DrFaust92 Friendly ping 😄 , do you think the above use-cases are valid? Or should I remove the `CustomizeDiff` function?